src/main/java/io/stargate/sgv2/jsonapi/service/operation/tables/WhereCQLClauseAnalyzer.java

```diff
@@ -239,6 +241,39 @@ private void checkFilteringOnComplexColumns(Map<CqlIdentifier, TableFilter> iden
     }
   }
 
+  /**
+   * Check if any non $eq filter is used.
+   *
+   * <p>e.g. Table UpdateOne and DeleteOne, DATA API does not know the exact modified row count in
```

Comment:
Isn't it also possible for `$eq` to match multiple rows? For example, would `deleteOne({ isDeleted: false })` delete multiple documents if multiple documents have `isDeleted: false`? Potentially silly question, I just wanted to make sure. Otherwise this PR looks reasonable.

Reply:
Yes, if we did not require filters to cover the whole primary key, but we do (for deletes and updates), I think. So only-eq-filters is a necessary but not sufficient constraints -- filtering on all primary key columns is also required.

Reply:
Thanks folks
Correct, for both deleteOne and updateOne, there is a rule checking (checkFullPrimaryKey) to enforce the full primary key is constraint in filter clause.
And yes, so `only-eq-filters is a necessary but not sufficient constraints`